staging/src/k8s.io/apimachinery/pkg/api/meta/conditions.go

```diff
@@ -57,7 +57,7 @@ func SetStatusCondition(conditions *[]metav1.Condition, newCondition metav1.Cond
 // RemoveStatusCondition removes the corresponding conditionType from conditions.
 // conditions must be non-nil.
 func RemoveStatusCondition(conditions *[]metav1.Condition, conditionType string) {
-	if conditions == nil {
+	if conditions == nil || len(*conditions) == 0 {
```

Comment:
LGTM

could you add a testcase to [TestRemoveStatusCondition](https://github.com/kubernetes/kubernetes/blob/c678434623be4957d892a9865e5649f887a40c49/staging/src/k8s.io/apimachinery/pkg/api/meta/conditions_test.go#L90)?